File: src/main/java/eu/mshade/mwork/ShrinkingUUID.java
package eu.mshade.mwork;

import java.security.SecureRandom;
import java.util.Base64;
import java.util.UUID;

public class ShrinkingUUID {

    private static final SecureRandom SECURE_RANDOM = new SecureRandom();

    public static String randomUUID(int size){

        byte[] bytes = new byte[size];
        SECURE_RANDOM.nextBytes(bytes);
        return Base64.getEncoder().encodeToString(bytes);
    }

    public static String randomUUID(){
        return randomUUID(16);
    }
}

File: src/main/java/eu/mshade/mwork/ShrinkingUUID.kt
package eu.mshade.mwork

import java.security.SecureRandom

object ShrinkingUUID {

    private val SECURE_RANDOM = SecureRandom()

    @JvmOverloads
    fun randomUUID(size: Int = 8): String {
        val bytes = ByteArray(size)
        SECURE_RANDOM.nextBytes(bytes)
        val stringBuilder = StringBuilder()
        for (b in bytes) {
            stringBuilder.append((b.toInt() and 0xFF).toString(16))
        }
        return stringBuilder.toString()
    }
}